### PR TITLE
[MOB-3409] hide locate me button when quotes appear

### PR DESCRIPTION
**Pull request by EdwardW2**
## JIRA
[MOB-3409](https://karhoo.atlassian.net/browse/MOB-3409)

## DESCRIPTION
[Create a description of what you have changed in this branch]

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [ ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [x] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [x] Lint and unit tests pass locally (if appropriate)
- [ ] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/map/BookingMapView.kt [hunk 1]
```diff
@@ -388,15 +388,15 @@ class BookingMapView @JvmOverloads constructor(context: Context,
     fun setNoBottomPadding() {
         googleMap?.setPadding(0, 0, 0, 0)
         recentreMapIfDestinationIsNull()
+        showLocateUserButton()
 
         animateLocateMeButton(R.dimen.spacing_small, R.integer.animation_duration_slide_out_or_in_quotes)
```
Comment (mohadian) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/booking/map/BookingMapView.kt:393]: Do we still need this animateLocateMeButton?

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/map/BookingMapView.kt [hunk 2]
```diff
@@ -388,15 +388,15 @@ class BookingMapView @JvmOverloads constructor(context: Context,
     fun setNoBottomPadding() {
         googleMap?.setPadding(0, 0, 0, 0)
         recentreMapIfDestinationIsNull()
+        showLocateUserButton()
```
Comment (mateid-kr) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/booking/map/BookingMapView.kt:391]: We could have a single method which hides and show the button

For example fun showLocationButton(show: Boolean) {
if(show) {

} else {
}
}

That way we optimize the number of methods both in the view and contract.

**Review by mateid-kr — CHANGES_REQUESTED** [review]

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/map/BookingMapView.kt (hunk 1 above)
Comment (EdwardW2) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/booking/map/BookingMapView.kt:393]: Yes. This makes sure the button is in the correct position each time it is shown when the quote view disappears.

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/map/BookingMapView.kt (hunk 2 above)
Comment (EdwardW2) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/booking/map/BookingMapView.kt:391]: I don't really get what you mean here?

Comment (EdwardW2) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/booking/map/BookingMapView.kt:391]: Nevermind. I got it now

**Review by mateid-kr — APPROVED** [review]

**Pull request merged**